### PR TITLE
EN-7303: tx simulation route

### PR DESCRIPTION
**Pull request by bogdan-rosianu**

### CODE REVIEW

File: data/transaction.go [hunk 1]
```diff
@@ -122,6 +123,27 @@ type ResponseTransaction struct {
 	Code  string                  `json:"code"`
 }
 
+// TransactionSimulationResponseData holds the results of a transaction's simulation
+type TransactionSimulationResults struct {
+	Status     core.TransactionStatus                         `json:"status,omitempty"`
+	FailReason string                                         `json:"failReason,omitempty"`
+	ScResults  map[string]*transaction.SmartContractResultApi `json:"scResults,omitempty"`
```
Comment (andreibancioiu) [inline data/transaction.go:130]: We might have small issues in the future with serializing this `map[string]`, but ok, can stay like this for the moment.

File: go.sum [hunk 2]
```diff
@@ -1,4 +1,12 @@
 cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
+cloud.google.com/go v0.31.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (andreibancioiu) [inline go.sum:2]: A go mod tidy perhaps? Or completely remove `go.sum`, then do a `go build`?

Comment (bogdan-rosianu) [reply go.sum:2]: done

Comment (bogdan-rosianu) [reply go.sum:2]: both of them. `go mod tidy` was already run before your comment so no change, so I removed `go.sum` and then I run a `go build`

File: data/transaction.go (hunk 1 above)
Comment (bogdan-rosianu) [reply data/transaction.go:130]: you mean that a slice would have been safer? is this related to the order?

**Review by sasurobert — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**